### PR TITLE
KER-154 Use HDS elements for login

### PR DESCRIPTION
**Pull request by timwessman**
# Use HDS elements for login
[https://helsinkisolutionoffice.atlassian.net/browse/KER-154](https://helsinkisolutionoffice.atlassian.net/browse/KER-154)

## Otsikko 2 - Selitys mitä tapahtuu PR:ssä

Kirjautuminen yläpalkissa on korvattu HDS buttonilla, ikoneilla ja pudotusvalikolla. Voi olla, että ensin tulee katselmoida ja mergata [KER-153](https://helsinkisolutionoffice.atlassian.net/browse/KER-153) ja sen jälkeen tämän paikkaa hieman hienosäätää. 

Kun uusi [HDS Navigation](https://hds.hel.fi/components/navigation) -komponentti pian valmistuu (HDS versioon 3.0.0), se tulee korvaamaan koko yläpalkin. 

[KER-153]: https://helsinkisolutionoffice.atlassian.net/browse/KER-153?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,6 +64,7 @@
     "express": "^4.17.1",
     "file-loader": "^6.2.0",
     "font-awesome": "^4.7.0",
+    "hds-react": "^2.13.0",
```
Comment (mikkojamG) [inline package.json:67]: Remember to commit yarn.lock too 🙂

**Review by mikkojamG — CHANGES_REQUESTED** [review]

**Comment by arpargo** [comment]
Screenshot 2023-04-14 at 15 24 24

Is this just my issue or is the scaling not working properly?

**Comment by timwessman** [comment]
> Screenshot 2023-04-14 at 15 24 24
> 
> Is this just my issue or is the scaling not working properly?

My bad. Now its fixed

**Comment by timwessman** [comment]
I updated Babel to 7 to make tests run.

File: package.json [hunk 2]
```diff
@@ -23,6 +23,14 @@
...
   },
   "dependencies": {
+    "@babel/core": "^7.0.0",
+    "@babel/plugin-proposal-class-properties": "^7.0.0",
+    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
+    "@babel/plugin-syntax-dynamic-import": "^7.0.0",
+    "@babel/polyfill": "^7.0.0",
+    "@babel/preset-env": "^7.0.0",
+    "@babel/preset-react": "^7.0.0",
+    "@babel/register": "^7.0.0",
```
Comment (mikkojamG) [inline package.json:33]: I'm having this error when trying to run build: `Error: Cannot find module 'babel-register'`, not sure why that is 🤔

**Review by mikkojamG — COMMENTED** [review]
Nice work fixing the tests 🎉 . Had some issues when trying to run build, one being with node-sass complaining not founding right python version. That can be fixed pretty easily by replacing `node-sass` with `sass` package. Had another issue with babel/register which I was not able to figure out yet 🤔

File: package.json (hunk 2 above)
Comment (timwessman) [reply package.json:33]: now I got same problem. I was pretty sure I fixed this already. Now I can't find a sollution.

Comment (mikkojamG) [reply package.json:33]: I think I found it in `compile/index.js`. Need to replace `require('babel-register');` -> `require('@babel/register');`

File: .babelrc [hunk 3]
```diff
@@ -1,19 +1,20 @@
 {
```
Comment (mikkojamG) [inline .babelrc:1]: Not directly related, but this file must have changed from `babel.json` to `.babelrc` at some point. Dockerfile still has references to babel.json, which breaks the build. 🤔

File: package.json [hunk 4]
```diff
@@ -144,27 +146,29 @@
     "webpack-merge": "^4.2.2"
   },
   "devDependencies": {
-    "babel-jest": "^16.0.0",
-    "babel-preset-flow": "^6.23.0",
+    "@babel/core": "^7.0.0",
```
Comment (mikkojamG) [inline package.json:149]: Duplicate dependency here 👀

File: package.json [hunk 5]
```diff
@@ -144,27 +146,29 @@
...
+    "@babel/preset-flow": "^7.0.0",
```
Comment (mikkojamG) [inline package.json:150]: This should be moved to `dependencies`, otherwise it breaks Docker build 🤔

File: package.json [hunk 6]
```diff
@@ -144,27 +146,29 @@
...
+    "babel-core": "^7.0.0-bridge.0",
```
Comment (mikkojamG) [inline package.json:151]: Duplicate dependency here 👀

**Review by mikkojamG — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (timwessman) [reply package.json:33]: yes, that was it

**Comment by timwessman** [comment]
I fixed the font size and removed border of logged in user button

**Review by mikkojamG — CHANGES_REQUESTED** [review]
Code wise looks very good 👌 Found a bug where something in login dropdown causes sidescrolling 🤔 

Screenshot 2023-04-28 at 8 31 02

**Review by mikkojamG — APPROVED** [review]

**Pull request merged**